app/models/recommendation.rb

```diff
@@ -25,6 +25,52 @@ def validate(record)
 
   before_validation :set_pending_status
 
+  def self.create_by_id_and_email(recommender, params)
+    recommendee_ids = params["recommendee_ids"]
+    recommendee_emails = params["recommendee_emails"]
+    item_class = Kernel.const_get(params['item_type'])
+    item = item_class.find(params['item_id'])
+
+    new_recommendations = []
+
+    unless recommendee_ids.nil?
+      recommendee_ids.each do |id|
+        recommendee = User.find(id)
+        reco = Recommendation.new(
+          :recommender => recommender,
+          :recommendee => recommendee,
+          :item => item
+        )
+        if reco.save
+          new_recommendations.append(reco)
+        else
+          #Handle batch error
+        end
+      end
+    end
+
+    unless recommendee_emails.nil?
+      recommendee_emails.each do |email|
+        recommendee = User.find_by_email(email)
+        if recommendee
+          reco = Recommendation.new(
+            :recommender => recommender,
+            :recommendee => recommendee,
+            :item => item
+          )
+          if reco.save
+            new_recommendations.append(reco)
+          else
+            #Handle batch error
+          end
+        else
+          #TODO - New user. Send email with new recommendation.
+        end
```

Comment:
Well I could refactor out the common stuff here. That oughtta make it shorter. But I don't want to go against [Rule 1](http://nathanmarz.com/blog/suffering-oriented-programming.html) for now :wink: